src/main/java/org/kizombadev/markdownparser/SyntaxAnalyzer.java

```diff
@@ -1,148 +1,233 @@
+            } else if (Token.DoubleNumberSign.equals(currentToken)) {
                 handleSmallHeadlineLine(root);
-            } else if (currentToken.equals(Token.GreaterThanSign)) {
+            } else if (Token.GreaterThanSign.equals(currentToken)) {
                 handleQuotation(root);
-            } else if (currentToken.equals(Token.DoubleStar)) {
-                handleBold(root);
-            } else if (currentToken.equals(Token.Star)) {
-                handleItalic(root);
-            } else if (currentToken.equals(Token.NewLine)) {
+            } else if (Token.Star.equals(currentToken) && Token.Blank.equals(nextToken())) {
+                handleUnorderedList(root);
+            } else if (Token.NewLine.equals(currentToken)) {
                 stepTokenForward();
+                blankCounter = 0;
+            } else {
+                handleParagraph(root);
             }
 
+            checkInfinityLoop();
+
             currentToken = currentToken();
         }
 
         return root;
     }
 
-    private void handleQuotation(RootSyntax currentRoot) {
-        Syntax quotation = new Quotation();
-        stepTokenForward();
+    private void handleLineContainer(Syntax currentRoot) {
+        SkipBlanks();
 
-        handleLineContainer(quotation);
-        currentRoot.addChild(quotation);
+        Token currentToken = currentToken();
+
+        while (currentToken != null && !currentToken.equals(Token.NewLine)) {
+
+            if ((Token.DoubleStar.equals(currentToken) && isBoldModeActive) ||
+                    (Token.Star.equals(currentToken) && isItalicModeEnabled)) {
+                return;
+            } else if (Token.DoubleStar.equals(currentToken)) {
+                handleBold(currentRoot);
+            } else if (Token.Star.equals(currentToken)) {
+                handleItalic(currentRoot);
+            } else if (currentToken.isTextToken()) {
+                currentRoot.addChild(Syntax.createWithContent(SyntaxType.TEXT, addBlanks() + currentToken.getTextValue()));
+                stepTokenForward();
+            } else if (Token.Blank.equals(currentToken)) {
+                stepTokenForward();
+                blankCounter++;
+            }
+
+            checkInfinityLoop();
+
+            currentToken = currentToken();
+        }
     }
 
-    private void handleBigHeadlineLine(Syntax currentRoot) {
-        Syntax bigHeadline = new BigHeadline();
-        stepTokenForward();
+    private void checkInfinityLoop() {
+        if (infinityLoopCounter > INFINITY_LOOP_DETECTION_COUNT) {
+            throw new MarkdownParserException();
+        }
 
-        handleLineContainer(bigHeadline);
-        currentRoot.addChild(bigHeadline);
+        infinityLoopCounter++;
     }
 
-    private void handleSmallHeadlineLine(Syntax currentRoot) {
-        Syntax smallHeadline = new SmallHeadline();
-        stepTokenForward();
+    private void SkipBlanks() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00100)